dev/PushNotifications/PushNotificationsLongRunningTask.StartupTask/winmain.cpp

```diff
@@ -0,0 +1,24 @@
+﻿#include <windows.h>
+#include "pch.h"
+
+#include "../PushNotifications-Constants.h"
+
+#include <NotificationsLongRunningProcess_h.h>
+#include <wil/resource.h>
+
+int WINAPI WinMain(HINSTANCE /*hInstance*/, HINSTANCE /*hPrevInstance*/, PSTR /*lpCmdLine*/, int /*nCmdShow*/)
+{
+    RETURN_IF_FAILED(::CoInitializeEx(nullptr, COINITBASE_MULTITHREADED));
+
+    auto scopeExit = wil::scope_exit(
+        [&]() { CoUninitialize(); });
+
+    try
+    {
+        wil::com_ptr<INotificationsLongRunningPlatform> longRunningProcessPlatform
+            { wil::CoCreateInstance<NotificationsLongRunningPlatform, INotificationsLongRunningPlatform>(CLSCTX_LOCAL_SERVER) };
+    }
+    CATCH_RETURN()
```

Comment:
Seems like there is no retry logic or restart logic here.  What is your reliability strategy for this LRP?

Reply:
And does CATCH_RETURN give you sufficient telemetry to assess reliability?

Reply:
I believe there are retryable RPC errors that we can use here @EHO-makai? My thinking is that we can retry a max of three times with exponential back-off and then give up

Reply:
Also @danielayala94 , This is an excellent spot to add a telemetry event to guage platform startup reliability.

Reply:
This is a great point. Right now, if startup task fails on creating the LRP, then it won't be launched. Assuming a full implementation of the push notification feature, our next chance to launch the LRP will happen if an unpackaged app calls into the SDK and that function needs to talk to the LRP. Such functionality is not implemented yet and has to be proposed on PRs opened against WNP_LRP (this PR's branch).

I'm wondering, how valuable is implement a retry logic for CoCreating the LRP? Are there any legit scenarios where we should try again immediately?

About telemetry, @loneursid did you verify if the telemetry for Windows App SDK is usable now? This is something we need to consider here and for the feature in general (though maybe not for 9/3?).

Reply:
I know shell had done some work in the past to implement an IsRPCRetryableError(Hresult). You may want to research this and implement something similar here.